packages/noir-ethereum-api/src/ethereum/createRecordingClient.ts

```diff
@@ -0,0 +1,51 @@
+import { PublicClient } from 'viem';
+
+export type CallResultEntry = {
```

Comment:
1) I propose just `Call`, as CallResult is basically `call.result`.
2) I would add annotation about promise to this type not the other one, for two reasons:
- it is hidden form user, not need to export it
- we can use word Promise, that is already well recognized
Hence I propose: `CallWithResultPromise`

Reply:
Renamed `CallResultEntry` -> `CallWithResultPromise` and `AwaitedCallResultEntry` to `Call` but IMHO it inconsistent. I think `Call` should be named `CallWithResult`